tools/data-api/internal/repositories/services.go

```diff
@@ -353,59 +164,244 @@ func (s *ServicesRepositoryImpl) ProcessResourceDefinitions(serviceName string,
 	resourceDefinition := ServiceApiVersionResourceDetails{}
 	resourceSchema := ResourceSchema{}
 	constants := make(map[string]ConstantDetails, 0)
+	models := make(map[string]ModelDetails)
+	operations := make(map[string]ResourceOperations)
+	resourceIds := make(map[string]ResourceIdDefinition)
 
-	// TODO rename this var so we can import path to use path.Join
-	path := fmt.Sprintf("%s/%s/%s/%s", s.directory, serviceName, version, resource)
-	files, err := os.ReadDir(path)
+	resourcePath := path.Join(s.directory, serviceName, version, resource)
```

Comment:
I think it's quite implicit that we're dealing with the API items/things here, are we trying to differentiate version and resource from other versions or resources within the data API? If not I actually find it confusing to append these with `api` since it makes me question what other possible types of version and resource there are.

Reply:
TBH this was mostly in reference to `resource` which is ambiguous between the Resource within an API Version and a Terraform Resource - `apiVersion` was mostly for consistency, since we'll likely end up reusing this in other functions.

Providing we stick with the terminology `resource` and `terraformResource` to differentiate those two that'd work fine 👍